frontend/src/js/components/settings/upgrade.js

```diff
@@ -13,53 +13,53 @@
 //    limitations under the License.
 import React, { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { useNavigate } from 'react-router-dom';
 
-import { InfoOutlined as InfoOutlinedIcon, LocalOffer as LocalOfferIcon } from '@mui/icons-material';
-
-import InfoText from '@northern.tech/common-ui/infotext';
-import Loader from '@northern.tech/common-ui/loader';
-import storeActions from '@northern.tech/store/actions';
-import { PLANS, TIMEOUTS } from '@northern.tech/store/constants';
+import { ConfirmUpgrade } from '@northern.tech/common-ui/dialogs/ConfirmUpgrade';
+import { PLANS } from '@northern.tech/store/constants';
 import { getFeatures, getOrganization } from '@northern.tech/store/selectors';
-import { completeUpgrade, getDeviceLimit, getUserOrganization, requestPlanChange } from '@northern.tech/store/thunks';
-import dayjs from 'dayjs';
+import { getUserOrganization, requestPlanChange } from '@northern.tech/store/thunks';
 
+import { EnterpriseRequestExpanded } from './EnterpriseRequestExpanded';
+import { PlanExpanded } from './PlanExpanded';
 import AddOnSelection from './addonselection';
-import CardSection from './cardsection';
 import PlanSelection from './planselection';
-import QuoteRequestForm from './quoterequestform';
-
-const { setSnackbar } = storeActions;
-
-const offerTag = (
-  <span className="offerTag">
-    <LocalOfferIcon /> End of year offer
-  </span>
-);
-
-export const PostUpgradeNote = ({ newPlan }) => (
-  <div style={{ maxWidth: 750 }} className="margin-top-small">
-    <h2 style={{ marginTop: 15 }}>Upgrade now</h2>
-    <div>
-      <p>
-        <b>Your upgrade was successful! </b>You are now signed up to the <b>{newPlan}</b> plan.
-      </p>
-      <p>Redirecting you to your organization page...</p>
-      <Loader show={true} />
-    </div>
-  </div>
-);
 
+export const updateLocalStorage = (orgId, name, isAdd) => {
+  const currentState = JSON.parse(localStorage.getItem(orgId + '_upgrades')) || {};
+  currentState[name] = { pending: true, isAdd };
+  localStorage.setItem(orgId + '_upgrades', JSON.stringify(currentState));
+};
+export const clearLocalStorageEntry = (orgId, name) => {
+  const currentState = JSON.parse(localStorage.getItem(orgId + '_upgrades')) || {};
+  delete currentState[name];
+  localStorage.setItem(orgId + '_upgrades', JSON.stringify(currentState));
+};
+export const isUpgrade = orgId => {
+  const currentState = JSON.parse(localStorage.getItem(orgId + '_upgrades')) || {};
+  if (currentState['enterprise']) {
```

Comment:
```suggestion
  if (currentState.enterprise) {
```
and below... or maybe get them in a list of plans to check and minimize duplication..